commit fcc5f22a8265a6f023c6ede2ee9e0f9691bbb1da (refs/changes/12/3212/2)
Author: Julien Gilli <julien.gilli@joyent.com>
Date:   2018-01-12T23:08:51+00:00 (1 year, 9 months ago)
    
    TRITON-46 sdc-cloudapi volumes-automount.test.js can stall when waiting for container to be deleted
    Reviewed by: Marsell Kukuljevic <marsell@joyent.com>
    Reviewed by: Mike Zeller <mike.zeller@joyent.com>
    Approved by: Marsell Kukuljevic <marsell@joyent.com>

diff --git a/test/volumes-automount.test.js b/test/volumes-automount.test.js
index 671407d..6def9f6 100644
--- a/test/volumes-automount.test.js
+++ b/test/volumes-automount.test.js
@@ -494,7 +494,7 @@ if (CONFIG.experimental_cloudapi_nfs_shared_volumes !== true) {
                         t.ok(machine.state, 'should have state, got: '
                             + machine.state + ' (try ' + attempt + ')');
                         state = machine.state;
-                        if (state === 'deleted') {
+                        if (state === 'deleted' || state === 'failed') {
                             t.end();
                         } else {
                             setTimeout(getState, 5000);
